parity/modules.rs

```diff
@@ -112,6 +113,37 @@ fn sync_arguments(io_path: &str, sync_cfg: SyncConfig, net_cfg: NetworkConfigura
 	BootArgs::new().stdin(service_payload).cli(cli_args)
 }
 
+#[cfg(feature="stratum")]
+pub fn stratum (hypervisor_ref: &mut Option<Hypervisor>, maybe_config: &Option<::ethcore::miner::StratumOptions>)
+{
+	use ethcore_stratum;
+
+	if let &Some(ref config) = maybe_config {
+		let mut hypervisor = hypervisor_ref.take().expect("There should be hypervisor for ipc configuration");
+		let args = BootArgs::new().stdin(
+				serialize(&ethcore_stratum::ServiceConfiguration {
+					io_path: hypervisor.io_path.to_owned(),
+					port: config.port,
+					listen_addr: config.listen_addr.to_owned(),
+					secret: config.secret,
+				}).expect("Any binary-derived struct is serializable by definition")
```

Comment:
side note: why isn't serialization infallible here? It requires `T: BinaryConvertable`. Usually only deserialization is fallible for sufficiently strong types.